scripts/extensions.sh

```diff
@@ -34,64 +47,38 @@ function enable_static_extension {
     local PHP=$1
     local EXTENSION=$2
 
-    echo "Enabling ${EXTENSION} extension"
+    echo "Enabling \"${EXTENSION}\" extension"
     phpenmod -v "${PHP}" -s ALL "${EXTENSION}"
 }
 
-function enable_sqlsrv {
-    local __result=$1
-    local PHP=$2
-    local -a EXTENSIONS=("${@:3}")
-    local TO_RETURN
-
-    if [[ ! ${PHP} =~ (7.3|7.4|8.0) ]];then
-        echo "Skipping enabling of sqlsrv extension; not supported on PHP < 7.3"
-    else
-        enable_static_extension "${PHP}" sqlsrv
-    fi
-
-    TO_RETURN=$(echo "${EXTENSIONS[@]}" | sed -E -e 's/php[0-9.]+-(pdo[_-]){0,1}sqlsrv/ /g' | sed -E -e 's/\s{2,}/ /g')
-    eval "$__result='$TO_RETURN'"
-}
+PHP=$1
+declare -a EXTENSIONS=("${@:2}")
+# shellcheck disable=SC2196
+ENABLED_EXTENSIONS=$(php -m | tr '[:upper:]' '[:lower:]' | egrep -v '^[\[]' | grep -v 'zend opcache')
+EXTENSIONS_TO_INSTALL=()
 
-function enable_swoole {
-    local __result=$1
-    local PHP=$2
-    local -a EXTENSIONS=("${@:3}")
-    local TO_RETURN
+# Loop through known statically compiled/installed extensions, and enable them.
+for EXTENSION in ${EXTENSIONS[@]}; do
```

Comment:
Based on what I know of bash, and a large part of that at this point in time has been via going through shellcheck error messages to see _why_ things are being flagged... quoting `${EXTENSIONS[@]}` here is the right thing to do, as what it does is quote _each entry_ in the array. This ensures that any elements of the array that have whitespace in them are treated as individual items, versus multiple items.

As an example, if  you defined:

```bash
EXTENSIONS=("foo" "foo bar" "bar")
```

then the following:

```bash
for EXTENSION in ${EXTENSIONS[@]}; do
    echo "${EXTENSION}"
done
```

would result in

```txt
foo
foo
bar
bar
```

while

```bash
for EXTENSION in "${EXTENSIONS[@]}";do
    echo "${EXTENSION}"
done
```

would result correctly in

```txt
foo
foo bar
bar
```

I recognize you saw a difference on Mac - I'm not sure why that is, but on the bash shipped with Ubuntu, the above is definitely true.

While we likely won't have those scenarios, this protects us from them.